app/controllers/statistics_controller.rb

```diff
@@ -1,23 +1,40 @@
+    @arch_count = Host.authorized(:view_hosts, Host).count_distribution :architecture
+    @env_count = Host.authorized(:view_hosts, Host).count_distribution :environment
+    @klass_count = Puppetclass.authorized(:view_puppetclasses).where('total_hosts>0').map { |pc| {:label => pc.to_label, :data => pc.total_hosts} }
+    @cpu_count = FactValue.authorized(:view_facts).my_facts.count_each "processorcount", :unit => Nn_('%s core', '%s cores')
     @model_count = FactValue.authorized(:view_facts).my_facts.count_each "manufacturer"
-    @mem_size    = FactValue.authorized(:view_facts).my_facts.mem_average "memorysize"
-    @mem_free    = FactValue.authorized(:view_facts).my_facts.mem_average "memoryfree"
-    @swap_size   = FactValue.authorized(:view_facts).my_facts.mem_average "swapsize"
-    @swap_free   = FactValue.authorized(:view_facts).my_facts.mem_average "swapfree"
+    @mem_size = FactValue.authorized(:view_facts).my_facts.mem_average "memorysize"
+    @mem_free = FactValue.authorized(:view_facts).my_facts.mem_average "memoryfree"
+    @swap_size = FactValue.authorized(:view_facts).my_facts.mem_average "swapsize"
+    @swap_free = FactValue.authorized(:view_facts).my_facts.mem_average "swapfree"
     respond_to do |format|
       format.html
       format.json do
-        render :json => { :statistics => { :os_count    => @os_count, :arch_count => @arch_count,
-                                           :env_count   => @env_count, :klass_count => @klass_count, :cpu_count => @cpu_count,
-                                           :model_count => @model_count, :mem_size => @mem_size, :mem_free => @mem_free, :swap_size => @swap_size,
-                                           :swap_free   => @swap_free, :mem_totsize => @mem_totsize, :mem_totfree => @mem_totfree } }
+        render :json => {:statistics => [
+            buildJSON("os_dist", "OS Distribution", @os_count, "hosts?search=os_title=~VAL~"),
+            buildJSON("arch_dist", "Architecture Distribution", @arch_count, "hosts?search=architecture=~VAL~"),
+            buildJSON("env_dist", "Environments Distribution", @env_count, "hosts?search=environment=~VAL~"),
+            buildJSON("cpu_num", "Number of CPUs", @cpu_count, "hosts?search=facts.processorcount=~VAL1~"),
+            buildJSON("hardware", "Hardware", @model_count, "hosts?search=facts.manufacturer~~VAL~"),
+            buildJSON("class_dist", "Class Distribution", @klass_count, "hosts?search=class=~VAL~")
+        #     buildJSON("mem_usage", "Average memory usage", [{:label=>_("free memory"), :data=>@mem_free},{:label=>_("used memory"),:data=>@mem_size-@mem_free}], options),
+        #     buildJSON("swap_usage", "Average swap usage", [{:label=>_("free swap"), :data=>@swap_free},{:label=>_("used swap"), :data=>@swap_size-@swap_free}], options)
+        ]
+        }
       end
     end
   end
+
+  def buildJSON (name, title, data, url)
```

Comment:
this method should be private.
also, build json sounds a bit generic, maybe chart data ?

and last, is this is something that is general for other charts too? if thats the case, maybe it should live in its own service class or something similar.